src/useCases/GetChecklistResultUseCase.ts

```diff
@@ -1,11 +1,26 @@
 import { pipe } from 'fp-ts/lib/function';
+import * as RA from 'fp-ts/ReadonlyArray';
 import * as TE from 'fp-ts/TaskEither';
 import { preLoadChecklist } from '../domain/checklist/preLoadChecklist';
-import { ChecklistResult, evalChecklist } from '../domain/checklist/types';
+import { ChecklistResult, evalChecklist, Result } from '../domain/checklist/types';
+import { uploadToS3Checklist } from '../domain/checklist/uploadToS3Checklist';
 import { PreLoadRecordRepository } from '../domain/PreLoadRepository';
+import { UploadToS3RecordRepository } from '../domain/UploadToS3RecordRepository';
 
 export const GetChecklistResultUseCase =
-  (preLoadRecordRepository: PreLoadRecordRepository) => (): TE.TaskEither<Error, ChecklistResult> =>
-    pipe(preLoadRecordRepository.list(), TE.map(evalChecklist(preLoadChecklist)));
+  (preLoadRecordRepository: PreLoadRecordRepository, uploadToS3RecordRepository: UploadToS3RecordRepository) =>
+  (): TE.TaskEither<Error, ChecklistResult> => {
+    const preLoadChecklistResult = pipe(preLoadRecordRepository.list(), TE.map(evalChecklist(preLoadChecklist)));
+    const uploadToS3ChecklistResult = pipe(
+      TE.of(RA.concatW),
+      TE.ap(preLoadRecordRepository.list()),
+      TE.ap(uploadToS3RecordRepository.list()),
+      TE.map(evalChecklist(uploadToS3Checklist))
+    );
```

Comment:
```suggestion
   const uploadToS3ChecklistResult = pipe(
      uploadToS3RecordRepository.list(),
      TE.map(evalChecklist(uploadToS3Checklist))
    );
```

Reply:
This suggestion makes me think that the structure should be improved so that the caller can't call this method without `PreLoadRecord` entities. At the moment is up to the knowledge of the developer :/

Reply:
Same here, once we enable the [strict](https://www.typescriptlang.org/tsconfig#strict) option this code should be changed because it will not compile anymore. Moreover we should improve the coverage of this use case 🤔